timeinterval/timeinterval.go

```diff
@@ -0,0 +1,213 @@
+package timeinterval
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var regexTimeStringISO = regexp.MustCompile("^(-?(?:[1-9][0-9]*)?[0-9]{4})-(1[0-2]|0[1-9])-(3[01]|0[1-9]|[12][0-9])T(2[0-3]|[01][0-9]):([0-5][0-9]):([0-5][0-9])(\\.[0-9]+)?(Z)?$")
+
+type formatType uint8
+
+// TypeUnknown indicates that the given string has a format that is unknown and unsupported.
+const TypeUnknown formatType = 0
+// TypeTime indicates that the given string is an ISO8601 time string
+const TypeTime formatType = 1
+// TypeDuration indicates that the given string is am ISO8601 duration string
+const TypeDuration formatType = 2
+
+const durationWeek = 7 * 24 * time.Hour
+const durationDay = 24 * time.Hour
+
+// TimeInterval is an interface that describes the API supported
+// for all intervals implemented in the timeinterval package.
+type TimeInterval interface {
```

Comment:
You don't need to export an interface.  If people need an interface they can do one themselves - that is the power of Go interfaces :)